<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>RPhenoscape Tutorial • SCATE.shortcourse</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="RPhenoscape Tutorial">
<meta property="og:description" content="SCATE.shortcourse">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SCATE.shortcourse</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/01_rphenoscape.html">RPhenoscape Tutorial</a>
    </li>
    <li>
      <a href="../articles/PARAMO.html">PARAMO Tutorial</a>
    </li>
    <li>
      <a href="../articles/glossary.html">Glossary</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/phenoscape/scate-shortcourse/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>RPhenoscape Tutorial</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/phenoscape/scate-shortcourse/blob/HEAD/vignettes/01_rphenoscape.Rmd" class="external-link"><code>vignettes/01_rphenoscape.Rmd</code></a></small>
      <div class="hidden name"><code>01_rphenoscape.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">1. Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The desired input of many comparative studies is a matrix that
includes all the data available for a set of traits and a focal taxon.
Bringing similar types of trait data together across studies is
notoriously difficult, but is straightforward with ontology annotations.
Trait data that are tagged using similar ontology terms can be
automatically called using R functions.</p>
<p>This workshop uses trait data for vertebrates that has been annotated
(tagged) with ontology terms. These data are accessed from the <a href="http://kb.phenoscape.org/" class="external-link">Phenoscape KB</a>. In this lesson we
use R functions to obtain and view character matrices from the KB,
obtain synthetic matrices of inferred presence/absence characters, and
understand the meaning and relations of ontology terms. The lesson
assumes basic knowledge of R and RStudio.</p>
</div>
<div class="section level2">
<h2 id="browse-and-filter-data-in-the-kb-to-understand-the-scope-of-available-data-">2. Browse and filter data in the KB to understand the scope of
available data.<a class="anchor" aria-label="anchor" href="#browse-and-filter-data-in-the-kb-to-understand-the-scope-of-available-data-"></a>
</h2>
<p>You would like to understand the evolution of morphological traits in
a taxonomic group, and you need a matrix to map to your (typically
molecular) phylogenetic tree. You come to the Phenoscape KB to see what
types of data are available across the members of your clade. In this
exercise, you are an ichthyologist, interested in catfishes
(Siluriformes), specifically the bullhead catfishes in the family
Ictaluridae.</p>
<p>You begin by querying ‘Siluriformes’ in the Phenoscape KB <a href="https://kb.phenoscape.org/facet" class="external-link">faceted browsing page</a>. You
immediately see the scope of the data:</p>
<p>From left to right across the top tabs, for ‘Siluriformes’:</p>
<ul>
<li>There are 4,969 unique phenotypes</li>
<li>There are 292,853 taxon annotations</li>
<li>There are 1,505 siluriform taxa with data in the KB</li>
<li>There are 40 publications in the KB that include catfish
phenotypes</li>
<li>There are 19,232 candidate genes linked to the phenotypes</li>
</ul>
<p>Taxon annotations are the ontology terms tagged to original free text
descriptions of phenotypes for each siluriform taxon. Click on the
‘Taxon annotations’ tab, and then the ‘Sources’ box on the right hand
side of each row to see the publication source(s) and free text
description of the phenotype to which each ontology annotation was
tagged.</p>
<p>You are specifically interested in Ictaluridae (bullhead catfishes),
and you click on that family in the faceted browsing interface. Now
filter using ‘fin’ as the anatomical entity under ‘Query’ and include
parts (check the ‘parts’ box.)</p>
<p>Download these taxa and their morphological trait data using the link
provided.</p>
<p>Go to the publications tab, which shows nine publications that
contain fin data. Note that any of these studies, e.g., Lundberg (1992),
can be entered under ‘publications’ in faceted browsing, but the
original matrix cannot be downloaded.</p>
</div>
<div class="section level2">
<h2 id="retrieve-term-info-using-rphenoscape">3. Retrieve term info using RPhenoscape<a class="anchor" aria-label="anchor" href="#retrieve-term-info-using-rphenoscape"></a>
</h2>
<p>Now we will repeat some of the above steps in R, beginning with a
query for ‘Ictaluridae’ and ‘fin’ (including parts) to retrieve the list
of studies that contain fin characters for this taxon.</p>
<p>The <a href="http://rphenoscape.phenoscape.org/" class="external-link">RPhenoscape</a>
package provides convenient access to the Phenoscape Knowledgebase in
the form of R functions returning R-native data structures. So that we
don’t have to prefix every call of the package’s functions with the
package name, we will first attach the package:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/phenoscape/rphenoscape" class="external-link">rphenoscape</a></span><span class="op">)</span></span></code></pre></div>
<p>First let’s look up information about the query terms. The <a href="https://rphenoscape.phenoscape.org/reference/find_term.html" class="external-link"><code>find_term()</code></a>
function searches the KB for terms matching a given search text, and
will return the ID (term IRI), label, source ontology IRI, and type of
match (e.g., exact, broad). For example, search for terms containing the
string “fin” in the label or synonym:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="http://rphenoscape.phenoscape.org/reference/find_term.html" class="external-link">find_term</a></span><span class="op">(</span><span class="st">"fin"</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                               id</span></span>
<span><span class="co">#&gt; 1  http://purl.obolibrary.org/obo/UBERON_0008897</span></span>
<span><span class="co">#&gt; 2  http://purl.obolibrary.org/obo/UBERON_0002389</span></span>
<span><span class="co">#&gt; 3  http://purl.obolibrary.org/obo/UBERON_0009565</span></span>
<span><span class="co">#&gt; 4  http://purl.obolibrary.org/obo/UBERON_0009552</span></span>
<span><span class="co">#&gt; 5     http://purl.obolibrary.org/obo/ZFA_0000108</span></span>
<span><span class="co">#&gt; 6  http://purl.obolibrary.org/obo/UBERON_2005316</span></span>
<span><span class="co">#&gt; 7      http://purl.obolibrary.org/obo/GO_0033333</span></span>
<span><span class="co">#&gt; 8      http://purl.obolibrary.org/obo/GO_0033334</span></span>
<span><span class="co">#&gt; 9      http://purl.obolibrary.org/obo/GO_0031101</span></span>
<span><span class="co">#&gt; 10     http://purl.obolibrary.org/obo/HP_0001231</span></span>
<span><span class="co">#&gt; 11     http://purl.obolibrary.org/obo/HP_0001161</span></span>
<span><span class="co">#&gt; 12     http://purl.obolibrary.org/obo/HP_0001187</span></span>
<span><span class="co">#&gt; 13     http://purl.obolibrary.org/obo/HP_0001238</span></span>
<span><span class="co">#&gt; 14     http://purl.obolibrary.org/obo/HP_0001166</span></span>
<span><span class="co">#&gt; 15     http://purl.obolibrary.org/obo/HP_0001182</span></span>
<span><span class="co">#&gt; 16     http://purl.obolibrary.org/obo/HP_0001817</span></span>
<span><span class="co">#&gt; 17     http://purl.obolibrary.org/obo/HP_0001500</span></span>
<span><span class="co">#&gt; 18     http://purl.obolibrary.org/obo/HP_0001821</span></span>
<span><span class="co">#&gt; 19     http://purl.obolibrary.org/obo/HP_0004095</span></span>
<span><span class="co">#&gt; 20     http://purl.obolibrary.org/obo/HP_0004097</span></span>
<span><span class="co">#&gt; 21     http://purl.obolibrary.org/obo/HP_0002213</span></span>
<span><span class="co">#&gt; 22     http://purl.obolibrary.org/obo/HP_0004099</span></span>
<span><span class="co">#&gt; 23     http://purl.obolibrary.org/obo/HP_0006101</span></span>
<span><span class="co">#&gt; 24     http://purl.obolibrary.org/obo/HP_0006107</span></span>
<span><span class="co">#&gt; 25     http://purl.obolibrary.org/obo/HP_0001812</span></span>
<span><span class="co">#&gt; 26     http://purl.obolibrary.org/obo/HP_0001804</span></span>
<span><span class="co">#&gt; 27     http://purl.obolibrary.org/obo/HP_0005867</span></span>
<span><span class="co">#&gt; 28     http://purl.obolibrary.org/obo/HP_0007477</span></span>
<span><span class="co">#&gt; 29     http://purl.obolibrary.org/obo/HP_0009380</span></span>
<span><span class="co">#&gt; 30     http://purl.obolibrary.org/obo/HP_0008391</span></span>
<span><span class="co">#&gt; 31     http://purl.obolibrary.org/obo/HP_0010525</span></span>
<span><span class="co">#&gt; 32     http://purl.obolibrary.org/obo/HP_0009700</span></span>
<span><span class="co">#&gt; 33     http://purl.obolibrary.org/obo/HP_0009381</span></span>
<span><span class="co">#&gt; 34     http://purl.obolibrary.org/obo/HP_0010557</span></span>
<span><span class="co">#&gt; 35     http://purl.obolibrary.org/obo/HP_0008402</span></span>
<span><span class="co">#&gt; 36     http://purl.obolibrary.org/obo/HP_0010554</span></span>
<span><span class="co">#&gt; 37     http://purl.obolibrary.org/obo/HP_0025131</span></span>
<span><span class="co">#&gt; 38     http://purl.obolibrary.org/obo/HP_0025261</span></span>
<span><span class="co">#&gt; 39     http://purl.obolibrary.org/obo/HP_0011300</span></span>
<span><span class="co">#&gt; 40     http://purl.obolibrary.org/obo/HP_0031090</span></span>
<span><span class="co">#&gt; 41     http://purl.obolibrary.org/obo/HP_0030367</span></span>
<span><span class="co">#&gt; 42     http://purl.obolibrary.org/obo/HP_0030837</span></span>
<span><span class="co">#&gt; 43     http://purl.obolibrary.org/obo/HP_0030771</span></span>
<span><span class="co">#&gt; 44     http://purl.obolibrary.org/obo/HP_0030033</span></span>
<span><span class="co">#&gt; 45     http://purl.obolibrary.org/obo/HP_0030029</span></span>
<span><span class="co">#&gt; 46     http://purl.obolibrary.org/obo/HP_0012742</span></span>
<span><span class="co">#&gt; 47     http://purl.obolibrary.org/obo/HP_0012276</span></span>
<span><span class="co">#&gt; 48     http://purl.obolibrary.org/obo/HP_0100759</span></span>
<span><span class="co">#&gt; 49     http://purl.obolibrary.org/obo/HP_0034037</span></span>
<span><span class="co">#&gt; 50     http://purl.obolibrary.org/obo/HP_0045089</span></span>
<span><span class="co">#&gt; 51     http://purl.obolibrary.org/obo/HP_0100798</span></span>
<span><span class="co">#&gt; 52     http://purl.obolibrary.org/obo/HP_0040019</span></span>
<span><span class="co">#&gt; 53     http://purl.obolibrary.org/obo/HP_0100807</span></span>
<span><span class="co">#&gt; 54     http://purl.obolibrary.org/obo/HP_0045090</span></span>
<span><span class="co">#&gt; 55     http://purl.obolibrary.org/obo/HP_0033976</span></span>
<span><span class="co">#&gt; 56     http://purl.obolibrary.org/obo/MP_0000561</span></span>
<span><span class="co">#&gt; 57     http://purl.obolibrary.org/obo/MP_0000562</span></span>
<span><span class="co">#&gt; 58     http://purl.obolibrary.org/obo/MP_0003807</span></span>
<span><span class="co">#&gt; 59     http://purl.obolibrary.org/obo/MP_0000564</span></span>
<span><span class="co">#&gt; 60     http://purl.obolibrary.org/obo/MP_0002544</span></span>
<span><span class="co">#&gt; 61     http://purl.obolibrary.org/obo/MP_0006296</span></span>
<span><span class="co">#&gt; 62     http://purl.obolibrary.org/obo/MP_0009930</span></span>
<span><span class="co">#&gt; 63     http://purl.obolibrary.org/obo/MP_0030929</span></span>
<span><span class="co">#&gt; 64     http://purl.obolibrary.org/obo/MP_0030930</span></span>
<span><span class="co">#&gt; 65 http://purl.obolibrary.org/obo/UBERON_4000172</span></span>
<span><span class="co">#&gt; 66 http://purl.obolibrary.org/obo/UBERON_4400005</span></span>
<span><span class="co">#&gt; 67    http://purl.obolibrary.org/obo/ZFA_0001058</span></span>
<span><span class="co">#&gt; 68    http://purl.obolibrary.org/obo/VTO_0013147</span></span>
<span><span class="co">#&gt; 69    http://purl.obolibrary.org/obo/VTO_0003107</span></span>
<span><span class="co">#&gt; 70    http://purl.obolibrary.org/obo/VTO_0021975</span></span>
<span><span class="co">#&gt; 71    http://purl.obolibrary.org/obo/VTO_0018051</span></span>
<span><span class="co">#&gt; 72    http://purl.obolibrary.org/obo/VTO_0016713</span></span>
<span><span class="co">#&gt; 73    http://purl.obolibrary.org/obo/VTO_0015894</span></span>
<span><span class="co">#&gt; 74    http://purl.obolibrary.org/obo/VTO_0023422</span></span>
<span><span class="co">#&gt; 75    http://purl.obolibrary.org/obo/VTO_0023456</span></span>
<span><span class="co">#&gt; 76    http://purl.obolibrary.org/obo/VTO_0028227</span></span>
<span><span class="co">#&gt; 77    http://purl.obolibrary.org/obo/VTO_0023261</span></span>
<span><span class="co">#&gt; 78    http://purl.obolibrary.org/obo/VTO_0025094</span></span>
<span><span class="co">#&gt; 79    http://purl.obolibrary.org/obo/VTO_0023240</span></span>
<span><span class="co">#&gt; 80    http://purl.obolibrary.org/obo/VTO_0024763</span></span>
<span><span class="co">#&gt; 81    http://purl.obolibrary.org/obo/VTO_0029180</span></span>
<span><span class="co">#&gt; 82    http://purl.obolibrary.org/obo/VTO_0031838</span></span>
<span><span class="co">#&gt; 83    http://purl.obolibrary.org/obo/VTO_0059371</span></span>
<span><span class="co">#&gt; 84    http://purl.obolibrary.org/obo/VTO_0059142</span></span>
<span><span class="co">#&gt; 85    http://purl.obolibrary.org/obo/VTO_0063879</span></span>
<span><span class="co">#&gt; 86    http://purl.obolibrary.org/obo/VTO_0062623</span></span>
<span><span class="co">#&gt; 87    http://purl.obolibrary.org/obo/VTO_0073037</span></span>
<span><span class="co">#&gt; 88    http://purl.obolibrary.org/obo/VTO_0066755</span></span>
<span><span class="co">#&gt; 89    http://purl.obolibrary.org/obo/VTO_0069689</span></span>
<span><span class="co">#&gt; 90    http://purl.obolibrary.org/obo/VTO_0068161</span></span>
<span><span class="co">#&gt; 91    http://purl.obolibrary.org/obo/VTO_0073687</span></span>
<span><span class="co">#&gt;                                      label</span></span>
<span><span class="co">#&gt; 1                                      fin</span></span>
<span><span class="co">#&gt; 2                             manual digit</span></span>
<span><span class="co">#&gt; 3                     nail of manual digit</span></span>
<span><span class="co">#&gt; 4           distal segment of manual digit</span></span>
<span><span class="co">#&gt; 5                          fin (zebrafish)</span></span>
<span><span class="co">#&gt; 6                fin fold pectoral fin bud</span></span>
<span><span class="co">#&gt; 7                          fin development</span></span>
<span><span class="co">#&gt; 8                        fin morphogenesis</span></span>
<span><span class="co">#&gt; 9                         fin regeneration</span></span>
<span><span class="co">#&gt; 10          Abnormal fingernail morphology</span></span>
<span><span class="co">#&gt; 11                        Hand polydactyly</span></span>
<span><span class="co">#&gt; 12 Hyperextensibility of the finger joints</span></span>
<span><span class="co">#&gt; 13                          Slender finger</span></span>
<span><span class="co">#&gt; 14                          Arachnodactyly</span></span>
<span><span class="co">#&gt; 15                          Tapered finger</span></span>
<span><span class="co">#&gt; 16                       Absent fingernail</span></span>
<span><span class="co">#&gt; 17                            Broad finger</span></span>
<span><span class="co">#&gt; 18                              Broad nail</span></span>
<span><span class="co">#&gt; 19                          Curved fingers</span></span>
<span><span class="co">#&gt; 20                     Deviation of finger</span></span>
<span><span class="co">#&gt; 21                               Fine hair</span></span>
<span><span class="co">#&gt; 22                            Macrodactyly</span></span>
<span><span class="co">#&gt; 23                       Finger syndactyly</span></span>
<span><span class="co">#&gt; 24                Fingerpad telangiectases</span></span>
<span><span class="co">#&gt; 25                 Hyperconvex fingernails</span></span>
<span><span class="co">#&gt; 26                  Hypoplastic fingernail</span></span>
<span><span class="co">#&gt; 27               4-5 metacarpal synostosis</span></span>
<span><span class="co">#&gt; 28                Abnormal dermatoglyphics</span></span>
<span><span class="co">#&gt; 29                  Aplasia of the fingers</span></span>
<span><span class="co">#&gt; 30                  Dystrophic fingernails</span></span>
<span><span class="co">#&gt; 31                          Finger agnosia</span></span>
<span><span class="co">#&gt; 32                    Finger symphalangism</span></span>
<span><span class="co">#&gt; 33                            Short finger</span></span>
<span><span class="co">#&gt; 34                     Overlapping fingers</span></span>
<span><span class="co">#&gt; 35                       Ridged fingernail</span></span>
<span><span class="co">#&gt; 36             Cutaneous finger syndactyly</span></span>
<span><span class="co">#&gt; 37                         Finger swelling</span></span>
<span><span class="co">#&gt; 38                            Stiff finger</span></span>
<span><span class="co">#&gt; 39                         Broad fingertip</span></span>
<span><span class="co">#&gt; 40                       Finger dactylitis</span></span>
<span><span class="co">#&gt; 41                    Finger hyperphalangy</span></span>
<span><span class="co">#&gt; 42                             Finger pain</span></span>
<span><span class="co">#&gt; 43                           Mallet finger</span></span>
<span><span class="co">#&gt; 44                            Small finger</span></span>
<span><span class="co">#&gt; 45                         Splayed fingers</span></span>
<span><span class="co">#&gt; 46                         Thin fingernail</span></span>
<span><span class="co">#&gt; 47            Digital flexor tenosynovitis</span></span>
<span><span class="co">#&gt; 48                     Clubbing of fingers</span></span>
<span><span class="co">#&gt; 49            Pseudo-chilblains on fingers</span></span>
<span><span class="co">#&gt; 50                     Distinctive finding</span></span>
<span><span class="co">#&gt; 51                    Fingernail dysplasia</span></span>
<span><span class="co">#&gt; 52                     Finger clinodactyly</span></span>
<span><span class="co">#&gt; 53                            Long fingers</span></span>
<span><span class="co">#&gt; 54                           Minor finding</span></span>
<span><span class="co">#&gt; 55                        Volar fingernail</span></span>
<span><span class="co">#&gt; 56                                adactyly</span></span>
<span><span class="co">#&gt; 57                             polydactyly</span></span>
<span><span class="co">#&gt; 58                           camptodactyly</span></span>
<span><span class="co">#&gt; 59                              syndactyly</span></span>
<span><span class="co">#&gt; 60                           brachydactyly</span></span>
<span><span class="co">#&gt; 61                          arachnodactyly</span></span>
<span><span class="co">#&gt; 62                              fuzzy hair</span></span>
<span><span class="co">#&gt; 63            increased digit pigmentation</span></span>
<span><span class="co">#&gt; 64            decreased digit pigmentation</span></span>
<span><span class="co">#&gt; 65                          lepidotrichium</span></span>
<span><span class="co">#&gt; 66                                 fin ray</span></span>
<span><span class="co">#&gt; 67                  caudal fin (zebrafish)</span></span>
<span><span class="co">#&gt; 68                Callosciurus finlaysonii</span></span>
<span><span class="co">#&gt; 69                      Limnonectes finchi</span></span>
<span><span class="co">#&gt; 70                            Agama finchi</span></span>
<span><span class="co">#&gt; 71                    Plestiodon finitimus</span></span>
<span><span class="co">#&gt; 72                     Uroplatus finiavana</span></span>
<span><span class="co">#&gt; 73                         Varanus finschi</span></span>
<span><span class="co">#&gt; 74                         Amazona finschi</span></span>
<span><span class="co">#&gt; 75                        Aratinga finschi</span></span>
<span><span class="co">#&gt; 76                        Euphonia finschi</span></span>
<span><span class="co">#&gt; 77                    Micropsitta finschii</span></span>
<span><span class="co">#&gt; 78                       Oenanthe finschii</span></span>
<span><span class="co">#&gt; 79                     Psittacula finschii</span></span>
<span><span class="co">#&gt; 80                   Pycnonotus finlaysoni</span></span>
<span><span class="co">#&gt; 81                      Stizorhina finschi</span></span>
<span><span class="co">#&gt; 82                     Francolinus finschi</span></span>
<span><span class="co">#&gt; 83   Bunocephalus sp. (Fink and Fink 1981)</span></span>
<span><span class="co">#&gt; 84              Cnemidocarpa finmarkiensis</span></span>
<span><span class="co">#&gt; 85                       Garra findolabium</span></span>
<span><span class="co">#&gt; 86                         Schistura finis</span></span>
<span><span class="co">#&gt; 87                   Lamprologus finalimus</span></span>
<span><span class="co">#&gt; 88                  Saurenchelys finitimus</span></span>
<span><span class="co">#&gt; 89                        Soleonasus finis</span></span>
<span><span class="co">#&gt; 90                   Theragra finnmarchica</span></span>
<span><span class="co">#&gt; 91                   Benitochromis finleyi</span></span>
<span><span class="co">#&gt;                                  isDefinedBy matchType</span></span>
<span><span class="co">#&gt; 1  http://purl.obolibrary.org/obo/uberon.owl     exact</span></span>
<span><span class="co">#&gt; 2  http://purl.obolibrary.org/obo/uberon.owl     broad</span></span>
<span><span class="co">#&gt; 3  http://purl.obolibrary.org/obo/uberon.owl     broad</span></span>
<span><span class="co">#&gt; 4  http://purl.obolibrary.org/obo/uberon.owl     broad</span></span>
<span><span class="co">#&gt; 5     http://purl.obolibrary.org/obo/zfa.owl   partial</span></span>
<span><span class="co">#&gt; 6  http://purl.obolibrary.org/obo/uberon.owl   partial</span></span>
<span><span class="co">#&gt; 7      http://purl.obolibrary.org/obo/go.owl   partial</span></span>
<span><span class="co">#&gt; 8      http://purl.obolibrary.org/obo/go.owl   partial</span></span>
<span><span class="co">#&gt; 9      http://purl.obolibrary.org/obo/go.owl   partial</span></span>
<span><span class="co">#&gt; 10     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 11     http://purl.obolibrary.org/obo/hp.owl     broad</span></span>
<span><span class="co">#&gt; 12     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 13     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 14     http://purl.obolibrary.org/obo/hp.owl     broad</span></span>
<span><span class="co">#&gt; 15     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 16     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 17     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 18     http://purl.obolibrary.org/obo/hp.owl     broad</span></span>
<span><span class="co">#&gt; 19     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 20     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 21     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 22     http://purl.obolibrary.org/obo/hp.owl     broad</span></span>
<span><span class="co">#&gt; 23     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 24     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 25     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 26     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 27     http://purl.obolibrary.org/obo/hp.owl     broad</span></span>
<span><span class="co">#&gt; 28     http://purl.obolibrary.org/obo/hp.owl     broad</span></span>
<span><span class="co">#&gt; 29     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 30     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 31     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 32     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 33     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 34     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 35     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 36     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 37     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 38     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 39     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 40     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 41     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 42     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 43     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 44     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 45     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 46     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 47     http://purl.obolibrary.org/obo/hp.owl     broad</span></span>
<span><span class="co">#&gt; 48     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 49     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 50     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 51     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 52     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 53     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 54     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 55     http://purl.obolibrary.org/obo/hp.owl   partial</span></span>
<span><span class="co">#&gt; 56     http://purl.obolibrary.org/obo/mp.owl     broad</span></span>
<span><span class="co">#&gt; 57     http://purl.obolibrary.org/obo/mp.owl     broad</span></span>
<span><span class="co">#&gt; 58     http://purl.obolibrary.org/obo/mp.owl     broad</span></span>
<span><span class="co">#&gt; 59     http://purl.obolibrary.org/obo/mp.owl     broad</span></span>
<span><span class="co">#&gt; 60     http://purl.obolibrary.org/obo/mp.owl     broad</span></span>
<span><span class="co">#&gt; 61     http://purl.obolibrary.org/obo/mp.owl     broad</span></span>
<span><span class="co">#&gt; 62     http://purl.obolibrary.org/obo/mp.owl     broad</span></span>
<span><span class="co">#&gt; 63     http://purl.obolibrary.org/obo/mp.owl     broad</span></span>
<span><span class="co">#&gt; 64     http://purl.obolibrary.org/obo/mp.owl     broad</span></span>
<span><span class="co">#&gt; 65 http://purl.obolibrary.org/obo/uberon.owl     broad</span></span>
<span><span class="co">#&gt; 66 http://purl.obolibrary.org/obo/uberon.owl   partial</span></span>
<span><span class="co">#&gt; 67    http://purl.obolibrary.org/obo/zfa.owl   partial</span></span>
<span><span class="co">#&gt; 68    http://purl.obolibrary.org/obo/vto.owl   partial</span></span>
<span><span class="co">#&gt; 69    http://purl.obolibrary.org/obo/vto.owl   partial</span></span>
<span><span class="co">#&gt; 70    http://purl.obolibrary.org/obo/vto.owl   partial</span></span>
<span><span class="co">#&gt; 71    http://purl.obolibrary.org/obo/vto.owl   partial</span></span>
<span><span class="co">#&gt; 72    http://purl.obolibrary.org/obo/vto.owl   partial</span></span>
<span><span class="co">#&gt; 73    http://purl.obolibrary.org/obo/vto.owl   partial</span></span>
<span><span class="co">#&gt; 74    http://purl.obolibrary.org/obo/vto.owl   partial</span></span>
<span><span class="co">#&gt; 75    http://purl.obolibrary.org/obo/vto.owl   partial</span></span>
<span><span class="co">#&gt; 76    http://purl.obolibrary.org/obo/vto.owl   partial</span></span>
<span><span class="co">#&gt; 77    http://purl.obolibrary.org/obo/vto.owl   partial</span></span>
<span><span class="co">#&gt; 78    http://purl.obolibrary.org/obo/vto.owl   partial</span></span>
<span><span class="co">#&gt; 79    http://purl.obolibrary.org/obo/vto.owl   partial</span></span>
<span><span class="co">#&gt; 80    http://purl.obolibrary.org/obo/vto.owl   partial</span></span>
<span><span class="co">#&gt; 81    http://purl.obolibrary.org/obo/vto.owl   partial</span></span>
<span><span class="co">#&gt; 82    http://purl.obolibrary.org/obo/vto.owl   partial</span></span>
<span><span class="co">#&gt; 83    http://purl.obolibrary.org/obo/vto.owl   partial</span></span>
<span><span class="co">#&gt; 84    http://purl.obolibrary.org/obo/vto.owl   partial</span></span>
<span><span class="co">#&gt; 85    http://purl.obolibrary.org/obo/vto.owl   partial</span></span>
<span><span class="co">#&gt; 86    http://purl.obolibrary.org/obo/vto.owl   partial</span></span>
<span><span class="co">#&gt; 87    http://purl.obolibrary.org/obo/vto.owl   partial</span></span>
<span><span class="co">#&gt; 88    http://purl.obolibrary.org/obo/vto.owl   partial</span></span>
<span><span class="co">#&gt; 89    http://purl.obolibrary.org/obo/vto.owl   partial</span></span>
<span><span class="co">#&gt; 90    http://purl.obolibrary.org/obo/vto.owl   partial</span></span>
<span><span class="co">#&gt; 91    http://purl.obolibrary.org/obo/vto.owl   partial</span></span></code></pre></div>
<p>Note that because the KB integrates model organism phenotypes, the
results show types of fin specific to model organisms, such as “fin
(zebrafish)”.</p>
<p>There are several parameters available to narrow and filter search
results. For example, the parameter <code>matchType</code> allows
restricting to exact matches only:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="http://rphenoscape.phenoscape.org/reference/find_term.html" class="external-link">find_term</a></span><span class="op">(</span><span class="st">"fin"</span>, matchTypes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"exact"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                              id label</span></span>
<span><span class="co">#&gt; 1 http://purl.obolibrary.org/obo/UBERON_0008897   fin</span></span>
<span><span class="co">#&gt;                                 isDefinedBy matchType</span></span>
<span><span class="co">#&gt; 1 http://purl.obolibrary.org/obo/uberon.owl     exact</span></span></code></pre></div>
<p>To retrieve more details about a given term, such as definition,
synonyms and classification, you can pass the term IRI (e.g., ‘<a href="http://purl.obolibrary.org/obo/UBERON_0008897" class="external-link uri">http://purl.obolibrary.org/obo/UBERON_0008897</a>’ for fin)
to the function <a href="https://rphenoscape.phenoscape.org/reference/terminfo.html" class="external-link"><code>as.terminfo()</code></a>.
Classification details include the superclass (parent) and subclass
(child) relationships for a given ontology term, along with other
relationships such as part_of.</p>
<p>For example, view the relationships and term info for “pectoral
fin”:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pecfin</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://rphenoscape.phenoscape.org/reference/find_term.html" class="external-link">find_term</a></span><span class="op">(</span><span class="st">'pectoral fin'</span>, matchType<span class="op">=</span><span class="st">'exact'</span><span class="op">)</span></span>
<span><span class="fu"><a href="http://rphenoscape.phenoscape.org/reference/terminfo.html" class="external-link">as.terminfo</a></span><span class="op">(</span><span class="va">pecfin</span><span class="op">)</span></span>
<span><span class="co">#&gt; terminfo 'pectoral fin' http://purl.obolibrary.org/obo/UBERON_0000151</span></span>
<span><span class="co">#&gt; Definition: Paired fin that is located in the thoracic region of the body.</span></span>
<span><span class="co">#&gt; Synonyms:</span></span>
<span><span class="co">#&gt;     forefin (exact)</span></span>
<span><span class="co">#&gt;     pectoral fins (related)</span></span>
<span><span class="co">#&gt; Relationships:</span></span>
<span><span class="co">#&gt;     has skeleton pectoral fin skeleton</span></span>
<span><span class="co">#&gt;     develops from pectoral appendage bud</span></span>
<span><span class="co">#&gt; Subclass of:</span></span>
<span><span class="co">#&gt;     paired fin</span></span>
<span><span class="co">#&gt;     pectoral appendage</span></span>
<span><span class="co">#&gt; Superclass of:</span></span>
<span><span class="co">#&gt;     archipterygial fin</span></span>
<span><span class="co">#&gt;     pectoral fin (zebrafish)</span></span>
<span><span class="co">#&gt;     pectoral fin and (part_of some larva)</span></span>
<span><span class="co">#&gt;     pectoral fin and (part_of some larva)</span></span>
<span><span class="co">#&gt;     pectoral fin and (part_of some larval stage)</span></span>
<span><span class="co">#&gt;     pectoral fin and (part_of some larval stage)</span></span></code></pre></div>
<p>For a specific taxon, such as Siluriformes, you can also view the
taxon rank, whether or not the taxon is extinct, and its common
name:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">taxon</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://rphenoscape.phenoscape.org/reference/find_term.html" class="external-link">find_term</a></span><span class="op">(</span><span class="st">'Siluriformes'</span><span class="op">)</span> </span>
<span><span class="fu"><a href="http://rphenoscape.phenoscape.org/reference/terminfo.html" class="external-link">as.terminfo</a></span><span class="op">(</span><span class="va">taxon</span><span class="op">)</span></span>
<span><span class="co">#&gt; terminfo 'Siluriformes' http://purl.obolibrary.org/obo/VTO_0034991</span></span>
<span><span class="co">#&gt; Synonyms:</span></span>
<span><span class="co">#&gt;     catfish (related)</span></span>
<span><span class="co">#&gt;     catfishes (related)</span></span>
<span><span class="co">#&gt; Subclass of:</span></span>
<span><span class="co">#&gt;     Otophysi</span></span>
<span><span class="co">#&gt; Superclass of:</span></span>
<span><span class="co">#&gt;     Ailia</span></span>
<span><span class="co">#&gt;     Akysidae</span></span>
<span><span class="co">#&gt;     Amblycipitidae</span></span>
<span><span class="co">#&gt;     Amphiliidae</span></span>
<span><span class="co">#&gt;     Anchariidae</span></span>
<span><span class="co">#&gt;     Ariidae</span></span>
<span><span class="co">#&gt;     Aspredinidae</span></span>
<span><span class="co">#&gt;     Astroblepidae</span></span>
<span><span class="co">#&gt;     Auchenipteridae</span></span>
<span><span class="co">#&gt;     Auchenoglanididae</span></span>
<span><span class="co">#&gt;     Australoglanididae</span></span>
<span><span class="co">#&gt;     Austroglanididae</span></span>
<span><span class="co">#&gt;     Bagridae</span></span>
<span><span class="co">#&gt;     Callichthyidae</span></span>
<span><span class="co">#&gt;     Cetopsidae</span></span>
<span><span class="co">#&gt;     Chacidae</span></span>
<span><span class="co">#&gt;     Clariidae</span></span>
<span><span class="co">#&gt;     Claroteidae</span></span>
<span><span class="co">#&gt;     Clupisoma</span></span>
<span><span class="co">#&gt;     Conorhynchidae</span></span>
<span><span class="co">#&gt;     Cranoglanididae</span></span>
<span><span class="co">#&gt;     Diplomystidae</span></span>
<span><span class="co">#&gt;     Doradidae</span></span>
<span><span class="co">#&gt;     Eopeyeria</span></span>
<span><span class="co">#&gt;     Erethistidae</span></span>
<span><span class="co">#&gt;     Eutropiichthys</span></span>
<span><span class="co">#&gt;     Heptapteridae</span></span>
<span><span class="co">#&gt;     Heteropneustidae</span></span>
<span><span class="co">#&gt;     Horabagridae</span></span>
<span><span class="co">#&gt;     Hypophthalmidae</span></span>
<span><span class="co">#&gt;     Hypsidoridae</span></span>
<span><span class="co">#&gt;     Ictaluridae</span></span>
<span><span class="co">#&gt;     Lacantuniidae</span></span>
<span><span class="co">#&gt;     Laides</span></span>
<span><span class="co">#&gt;     Liobagrus</span></span>
<span><span class="co">#&gt;     Loricariidae</span></span>
<span><span class="co">#&gt;     Malapteruridae</span></span>
<span><span class="co">#&gt;     Mochokidae</span></span>
<span><span class="co">#&gt;     Nematogenyidae</span></span>
<span><span class="co">#&gt;     Neotropius</span></span>
<span><span class="co">#&gt;     Olyridae</span></span>
<span><span class="co">#&gt;     Pangasiidae</span></span>
<span><span class="co">#&gt;     Pareutropius</span></span>
<span><span class="co">#&gt;     Phreatobius</span></span>
<span><span class="co">#&gt;     Pimelodidae</span></span>
<span><span class="co">#&gt;     Platytropius</span></span>
<span><span class="co">#&gt;     Plotosidae</span></span>
<span><span class="co">#&gt;     Proeutropiichthys</span></span>
<span><span class="co">#&gt;     Pseudopimelodidae</span></span>
<span><span class="co">#&gt;     Rita</span></span>
<span><span class="co">#&gt;     Schilbidae</span></span>
<span><span class="co">#&gt;     Scoloplacidae</span></span>
<span><span class="co">#&gt;     Selenaspis</span></span>
<span><span class="co">#&gt;     Silonia</span></span>
<span><span class="co">#&gt;     Siluridae</span></span>
<span><span class="co">#&gt;     Siluroidei</span></span>
<span><span class="co">#&gt;     Sisoridae</span></span>
<span><span class="co">#&gt;     Trichomycteridae</span></span>
<span><span class="co">#&gt;     Xiurenbagrus</span></span>
<span><span class="co">#&gt; Extinct: FALSE</span></span>
<span><span class="co">#&gt; Rank: order</span></span>
<span><span class="co">#&gt; Common Name: catfish</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="retrieve-study-matrices-using-rphenoscape">4. Retrieve study matrices using RPhenoscape<a class="anchor" aria-label="anchor" href="#retrieve-study-matrices-using-rphenoscape"></a>
</h2>
<p>Now we will query for ‘Ictaluridae’ and ‘fin’ (including parts) to
retrieve the list of studies that contain fin characters for this
taxon.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">slist</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://rphenoscape.phenoscape.org/reference/get_studies.html" class="external-link">get_studies</a></span><span class="op">(</span>taxon <span class="op">=</span> <span class="st">"Ictaluridae"</span>, entity <span class="op">=</span> <span class="st">"fin"</span>, includeRels <span class="op">=</span> <span class="st">"part of"</span><span class="op">)</span></span></code></pre></div>
<p>The list is returned as a data.frame containing the study IDs and
citations. Navigate to Global Environment and click on ‘slist’ to view
the ID and citation (author, year) for each study in the KB that
contains any ictalurid species associated with fin data. Note that
taxon, entity, and quality are optional arguments in this query, and we
have also included fin parts. Here we found nine studies. Click on
‘slist’ in the Global Environment to view these studies and their
IDs.</p>
<p>Next we want to obtain the character matrix for one of these studies.
Using the study IDs from the previous step, we can get a list of the
character matrices as NeXML objects. We will choose the Lundberg (1992)
publication, the fifth item on the list above.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lundberg_nexml</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://rphenoscape.phenoscape.org/reference/get_study_data.html" class="external-link">get_study_data</a></span><span class="op">(</span><span class="va">slist</span><span class="op">$</span><span class="va">id</span><span class="op">[</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p>Now go to Global Environment pane and click on ‘lundberg_nexml’ to
view the study ID and associated NeXML object.</p>
<p>To view the full Lundberg (1992) matrix within RStudio, we need to
first retrieve the matrix as a data.frame from the NeXML object:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lundberg_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://rphenoscape.phenoscape.org/reference/get_char_matrix.html" class="external-link">get_char_matrix</a></span><span class="op">(</span><span class="va">lundberg_nexml</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, otus_id <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>You can take a look at a small part of the Lundberg (1992) matrix
(e.g., the first five taxa and first five characters):</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lundberg_matrix</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">7</span><span class="op">]</span></span>
<span><span class="co">#&gt;                 taxa                                      otu</span></span>
<span><span class="co">#&gt; 1  Ameiurus brunneus otu_331eacec-f62a-489b-aa97-01f88c469d4a</span></span>
<span><span class="co">#&gt; 2     Ameiurus catus otu_f0b4abb4-c4ca-4d99-b6d3-d069176eea82</span></span>
<span><span class="co">#&gt; 3     Ameiurus melas otu_5a3062f1-9803-4120-ae34-d2aa1ce02a75</span></span>
<span><span class="co">#&gt; 4   Ameiurus natalis otu_b92dfa0d-8c16-4228-815f-c418c6e92781</span></span>
<span><span class="co">#&gt; 5 Ameiurus nebulosus otu_31f42c8a-cd70-4bf4-994d-9bc483822ba7</span></span>
<span><span class="co">#&gt;   Anal-fin rays, species mean count Anterior dentations of pectoral spine</span></span>
<span><span class="co">#&gt; 1                                 1                                     3</span></span>
<span><span class="co">#&gt; 2                                 2                                     2</span></span>
<span><span class="co">#&gt; 3                                 2                                     1</span></span>
<span><span class="co">#&gt; 4                                 3                                     1</span></span>
<span><span class="co">#&gt; 5                                 2                                     2</span></span>
<span><span class="co">#&gt;   Anterior distal serrae of pectoral spine Anterior extent of sphenotic</span></span>
<span><span class="co">#&gt; 1                                        1                            0</span></span>
<span><span class="co">#&gt; 2                                        2                            0</span></span>
<span><span class="co">#&gt; 3                                        0                            0</span></span>
<span><span class="co">#&gt; 4                                        2                            0</span></span>
<span><span class="co">#&gt; 5                                        1                            0</span></span>
<span><span class="co">#&gt;   Anterior limb of fourth transverse process</span></span>
<span><span class="co">#&gt; 1                                          2</span></span>
<span><span class="co">#&gt; 2                                          1</span></span>
<span><span class="co">#&gt; 3                                          1</span></span>
<span><span class="co">#&gt; 4                                          0</span></span>
<span><span class="co">#&gt; 5                                          0</span></span></code></pre></div>
<p>Now navigate to Global Environment and click on ‘lundberg_matrix’ to
view the matrix, including the total number of taxa and characters
(i.e., the dimensions of the data.frame). To view the entire matrix in
this panel, click on the first icon that appears to the right when you
hover over the second row.</p>
<p>We can use the information in the NeXML object to convert character
state symbols to labels:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="http://rphenoscape.phenoscape.org/reference/state_symbols2labels.html" class="external-link">state_symbols2labels</a></span><span class="op">(</span><span class="va">lundberg_nexml</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, charmat <span class="op">=</span> <span class="va">lundberg_matrix</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">7</span><span class="op">]</span></span>
<span><span class="co">#&gt;                 taxa                                      otu</span></span>
<span><span class="co">#&gt; 1  Ameiurus brunneus otu_331eacec-f62a-489b-aa97-01f88c469d4a</span></span>
<span><span class="co">#&gt; 2     Ameiurus catus otu_f0b4abb4-c4ca-4d99-b6d3-d069176eea82</span></span>
<span><span class="co">#&gt; 3     Ameiurus melas otu_5a3062f1-9803-4120-ae34-d2aa1ce02a75</span></span>
<span><span class="co">#&gt; 4   Ameiurus natalis otu_b92dfa0d-8c16-4228-815f-c418c6e92781</span></span>
<span><span class="co">#&gt; 5 Ameiurus nebulosus otu_31f42c8a-cd70-4bf4-994d-9bc483822ba7</span></span>
<span><span class="co">#&gt;   Anal-fin rays, species mean count Anterior dentations of pectoral spine</span></span>
<span><span class="co">#&gt; 1                           16.1-20                                 large</span></span>
<span><span class="co">#&gt; 2                         20.1-25.5                              moderate</span></span>
<span><span class="co">#&gt; 3                         20.1-25.5                                 small</span></span>
<span><span class="co">#&gt; 4                           25.6-28                                 small</span></span>
<span><span class="co">#&gt; 5                         20.1-25.5                              moderate</span></span>
<span><span class="co">#&gt;   Anterior distal serrae of pectoral spine</span></span>
<span><span class="co">#&gt; 1               &lt;3 moderately sharp serrae</span></span>
<span><span class="co">#&gt; 2              3-6 moderately sharp serrae</span></span>
<span><span class="co">#&gt; 3             absent or scarcely developed</span></span>
<span><span class="co">#&gt; 4              3-6 moderately sharp serrae</span></span>
<span><span class="co">#&gt; 5               &lt;3 moderately sharp serrae</span></span>
<span><span class="co">#&gt;                                                    Anterior extent of sphenotic</span></span>
<span><span class="co">#&gt; 1 reaches or extends anterior to the level of anterior margin of epiphyseal bar</span></span>
<span><span class="co">#&gt; 2 reaches or extends anterior to the level of anterior margin of epiphyseal bar</span></span>
<span><span class="co">#&gt; 3 reaches or extends anterior to the level of anterior margin of epiphyseal bar</span></span>
<span><span class="co">#&gt; 4 reaches or extends anterior to the level of anterior margin of epiphyseal bar</span></span>
<span><span class="co">#&gt; 5 reaches or extends anterior to the level of anterior margin of epiphyseal bar</span></span>
<span><span class="co">#&gt;                     Anterior limb of fourth transverse process</span></span>
<span><span class="co">#&gt; 1 supracleithral facet shallow, basal recess shallow and broad</span></span>
<span><span class="co">#&gt; 2       supracleithral facet deep, basal recess deep and broad</span></span>
<span><span class="co">#&gt; 3       supracleithral facet deep, basal recess deep and broad</span></span>
<span><span class="co">#&gt; 4      supracleithral facet deep, basal recess deep and narrow</span></span>
<span><span class="co">#&gt; 5      supracleithral facet deep, basal recess deep and narrow</span></span></code></pre></div>
<p>Note that all taxa and characters from this study are returned, not
just those pertaining to our original search terms. In this example,
many anatomical entities other than ‘fin’ and its parts are returned as
part of the original study matrix. In a later step, we will learn how to
subset a matrix using anatomy or taxonomy terms.</p>
</div>
<div class="section level2">
<h2 id="get-a-presenceabsence-character-matrix-using-ontotrace-in-rphenoscape">5. Get a presence/absence character matrix using OntoTrace in
RPhenoscape<a class="anchor" aria-label="anchor" href="#get-a-presenceabsence-character-matrix-using-ontotrace-in-rphenoscape"></a>
</h2>
<p>In the previous step, we queried the KB for individual studies that
contain characters pertaining to the fins in ictalurid catfish, and we
obtained character matrices for source publications that included those
characters. Now we want to get a matrix that synthesizes these
characters into a single matrix.</p>
<p><a href="http://kb.phenoscape.org/ontotrace" class="external-link">OntoTrace</a> from the
KB combines characters from multiple publications into a single
synthetic matrix. It allows a user to specify any number or combination
of anatomical entities and taxa. It returns a matrix as a nexml object
that includes only presence or absence of anatomical entities. It uses
asserted data of presence or absence, i.e., originally described by an
author, or inferred data. Data are inferred from asserted characters,
e.g., if an author describes a part of a fin such as ‘fin ray’, the fin
is inferred present and scored as such in the matrix.</p>
<p>The OntoTrace default settings are to include parts of the anatomical
entity and to include variable-only characters, but here we will include
those that are invariant as well.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nex</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://rphenoscape.phenoscape.org/reference/get_ontotrace_data.html" class="external-link">get_ontotrace_data</a></span><span class="op">(</span>taxon <span class="op">=</span> <span class="st">"Ictaluridae"</span>, entity <span class="op">=</span> <span class="st">"fin"</span>, variable_only <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>To view the matrix within RStudio, get the matrix (m) as a
<code>data.frame</code> from the NeXML object. Opening this (‘m’) in the
Global Environment shows the taxa, character names, and state
assignments (0, absent; 1, present) (note: the otu and otus are
NeXML-internal IDs):</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://rphenoscape.phenoscape.org/reference/get_char_matrix.html" class="external-link">get_char_matrix</a></span><span class="op">(</span><span class="va">nex</span><span class="op">)</span></span></code></pre></div>
<p>The character and taxon lists for the matrix can also be returned.
Click on ‘meta’ in the Global Environment pane to view the data.frames
for taxa and entities:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://rphenoscape.phenoscape.org/reference/get_char_matrix_meta.html" class="external-link">get_char_matrix_meta</a></span><span class="op">(</span><span class="va">nex</span><span class="op">)</span></span></code></pre></div>
<p>The above returns 29 species of ictalurid catfishes and 124
anatomical entities for fins and their parts.</p>
<p>You will need to go to the <a href="http://kb.phenoscape.org/" class="external-link">Phenoscape KB</a> to view the
supporting states, i.e., the source data upon which inferences were made
or those that were directly asserted.</p>
<p>For example, in the returned OntoTrace matrix (see ‘m’ in Global
Environment (row 10)) the pelvic fin is scored as present (‘1’) for
<em>Ictalurus australis</em>. Enter these data on the <a href="https://kb.phenoscape.org/facet" class="external-link">faceted browsing page</a>:</p>
<ul>
<li>Anatomical entity = ‘pelvic fin’</li>
<li>Quality = ‘present’</li>
<li>Taxon = ‘Ictalurus australis’</li>
</ul>
<p>Looking at ‘Taxon annotations’, you will find that there are zero
phenotypes, i.e., no asserted character states to ‘pelvic fin present’.
Now click the ‘inferred presence’ radio button under Phenotypic quality.
You will now see two phenotypes for this taxon in the KB. These
phenotypes ‘pelvic fin lepidotrichium amount’ and ‘pelvic splint
present’ logically imply the presence of a pelvic fin. Thus, though an
author did not directly assert the presence of a pelvic fin in
<em>Ictalurus australis</em>, it is inferred and thus reported as
present (‘1’) in the OntoTrace matrix.</p>
</div>
<div class="section level2">
<h2 id="subset-the-matrix-taxonomically">6. Subset the matrix taxonomically<a class="anchor" aria-label="anchor" href="#subset-the-matrix-taxonomically"></a>
</h2>
<p>In the above step you retrieved a synthetic matrix for the
Ictaluridae. Now you would like to pare it down in taxonomic and
anatomical scope. Let’s begin by trimming the matrix down to the
presence and absence characters pertaining only to the genus
<em>Ictalurus</em>. We will reduce the matrix to include only
descendents of <em>Ictalurus</em>.</p>
<p>First, we need to parse the list to determine which taxa in the
matrix (‘m’) are descendants of <em>Ictalurus</em>. This returns a list
of TRUE or FALSE assignments for each taxon in the matrix:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">is_desc</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://rphenoscape.phenoscape.org/reference/is_relation.html" class="external-link">is_descendant</a></span><span class="op">(</span><span class="st">'Ictalurus'</span>, <span class="va">m</span><span class="op">$</span><span class="va">taxa</span><span class="op">)</span></span></code></pre></div>
<p>We then subset the matrix m to the descendants of
<em>Ictalurus</em>:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ictalurus_m</span> <span class="op">&lt;-</span> <span class="va">m</span><span class="op">[</span><span class="va">is_desc</span>, <span class="op">]</span></span></code></pre></div>
<p>View the new matrix by clicking on the ‘ictalurus_m’
<code>data.frame</code> in the Global Environment.</p>
<p>We can also subset the matrix by anatomy. In this case, let’s select
characters for the pectoral fin and its parts. We will use <a href="https://rphenoscape.phenoscape.org/reference/is_relation.html" class="external-link"><code>is_descendant()</code></a>
to first parse the characters (column names; note the first three
taxon-related columns are excluded) to determine which are descendents
(types or parts) of the term ‘pectoral fin’. Because this function only
looks for descendent terms, we also need to specify that any characters
pertaining to ‘pectoral fin’ itself are included.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">is_desc_pecfin</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://rphenoscape.phenoscape.org/reference/is_relation.html" class="external-link">is_descendant</a></span><span class="op">(</span><span class="st">'pectoral fin'</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">ictalurus_m</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span><span class="op">]</span>, includeRels <span class="op">=</span> <span class="st">"part_of"</span><span class="op">)</span> <span class="op">|</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">ictalurus_m</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span><span class="op">]</span> <span class="op">==</span> <span class="st">'pectoral fin'</span> </span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">is_desc_pecfin</span><span class="op">)</span></span>
<span><span class="co">#&gt; is_desc_pecfin</span></span>
<span><span class="co">#&gt; FALSE  TRUE </span></span>
<span><span class="co">#&gt;   110    14</span></span></code></pre></div>
<p>Using the ‘colnames’ command below, view the character names to
verify that pectoral fin related characters will be selected:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">ictalurus_m</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="va">is_desc_pecfin</span><span class="op">]</span></span>
<span><span class="co">#&gt;  [1] "anterior dentation of pectoral fin spine"       </span></span>
<span><span class="co">#&gt;  [2] "anterior distal serration of pectoral fin spine"</span></span>
<span><span class="co">#&gt;  [3] "pectoral fin"                                   </span></span>
<span><span class="co">#&gt;  [4] "pectoral fin lepidotrichium"                    </span></span>
<span><span class="co">#&gt;  [5] "pectoral fin proximal radial bone"              </span></span>
<span><span class="co">#&gt;  [6] "pectoral fin proximal radial cartilage"         </span></span>
<span><span class="co">#&gt;  [7] "pectoral fin proximal radial element"           </span></span>
<span><span class="co">#&gt;  [8] "pectoral fin radial bone"                       </span></span>
<span><span class="co">#&gt;  [9] "pectoral fin radial cartilage"                  </span></span>
<span><span class="co">#&gt; [10] "pectoral fin radial element"                    </span></span>
<span><span class="co">#&gt; [11] "pectoral fin radial skeleton"                   </span></span>
<span><span class="co">#&gt; [12] "pectoral fin skeleton"                          </span></span>
<span><span class="co">#&gt; [13] "pectoral fin spine"                             </span></span>
<span><span class="co">#&gt; [14] "posterior dentation of pectoral fin spine"</span></span></code></pre></div>
<p>We now subset the matrix m using ‘is_desc_pecfin’ and retain the
first three taxon-related columns:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">to_keep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="cn">TRUE</span>, <span class="cn">TRUE</span>, <span class="cn">TRUE</span>, <span class="va">is_desc_pecfin</span><span class="op">)</span> </span>
<span><span class="va">pecfin_m</span> <span class="op">&lt;-</span> <span class="va">ictalurus_m</span><span class="op">[</span>, <span class="va">to_keep</span><span class="op">]</span></span></code></pre></div>
<p>View this matrix of pectoral fin characters for <em>Ictalurus</em> by
clicking on ‘pecfin_m’ in the Global Environment.</p>
</div>
<div class="section level2">
<h2 id="retrieve-matrix-of-anatomical-dependencies">6. Retrieve matrix of anatomical dependencies<a class="anchor" aria-label="anchor" href="#retrieve-matrix-of-anatomical-dependencies"></a>
</h2>
<p>Knowing how the information about an anatomical structure may rely on
other layers of anatomical information is critical, if the goal is to
isolate and analyze independent features. We can query the KB for a
presence-absence dependency matrix for a specified set of anatomical
structures. The dependency matrix is derived from the formalized
knowledge of relationships among entities encoded in the ontology. In
the example below, the presence of ‘pelvic splint’ is dependent on the
presence of ‘pelvic fin’. In other words, the pelvic fin is implied
present when the pelvic splint is present because ‘pelvic splint’ is
part_of ‘pelvic fin’.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pectoral fin"</span>, <span class="st">"pelvic fin"</span>, <span class="st">"pelvic splint"</span>, <span class="st">"antorbital"</span>, <span class="st">"preopercle"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dep_mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://rphenoscape.phenoscape.org/reference/pa_dep_matrix.html" class="external-link">pa_dep_matrix</a></span><span class="op">(</span><span class="va">tl</span>, .names <span class="op">=</span> <span class="st">"label"</span><span class="op">)</span></span>
<span><span class="va">dep_mat</span></span>
<span><span class="co">#&gt;               pectoral fin pelvic fin pelvic splint antorbital preopercle</span></span>
<span><span class="co">#&gt; pectoral fin             1          0             0          0          0</span></span>
<span><span class="co">#&gt; pelvic fin               0          1             0          0          0</span></span>
<span><span class="co">#&gt; pelvic splint            0          1             1          0          0</span></span>
<span><span class="co">#&gt; antorbital               0          0             0          1          0</span></span>
<span><span class="co">#&gt; preopercle               0          0             0          0          1</span></span></code></pre></div>
<p>In the PARAMO tutorial, we will learn how to use these dependency
matrices to reconstruct individual traits and entire phenotypes.</p>
</div>
<div class="section level2">
<h2 id="retrieve-semantic-similarity-matrix">7. Retrieve semantic similarity matrix<a class="anchor" aria-label="anchor" href="#retrieve-semantic-similarity-matrix"></a>
</h2>
<p>We can also obtain similarity metrics for phenotypes based on
ontological distance. Three similarity metrics - Jaccard, Tanimoto, and
Cosine - are computed by the KB to measure similarity between terms (see
RPhenoscape documentation in the Help panel for how these are measured).
For example, here we compute and visualize Jaccard similarity for
premaxillary tooth phenotypes in Ictaluridae:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">phens</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://rphenoscape.phenoscape.org/reference/get_phenotypes.html" class="external-link">get_phenotypes</a></span><span class="op">(</span>entity <span class="op">=</span> <span class="st">"premaxillary tooth"</span>, taxon <span class="op">=</span> <span class="st">"Ictaluridae"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">phens</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 9</span></span>
<span><span class="va">sm</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://rphenoscape.phenoscape.org/reference/similarity.html" class="external-link">jaccard_similarity</a></span><span class="op">(</span>terms <span class="op">=</span> <span class="va">phens</span><span class="op">$</span><span class="va">id</span>, .labels <span class="op">=</span> <span class="va">phens</span><span class="op">$</span><span class="va">label</span>, .colnames <span class="op">=</span> <span class="st">"label"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/hclust.html" class="external-link">hclust</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/dist.html" class="external-link">as.dist</a></span><span class="op">(</span><span class="fl">1</span><span class="op">-</span><span class="va">sm</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="01_rphenoscape_files/figure-html/unnamed-chunk-20-1.png" width="700"></p>
<p>In the PARAMO tutorial, we’ll use a semantic similarity matrix to
visualize traits in a heat map. We are also developing methods that use
semantic similarity to cluster phenotypes into characters and character
states.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Hilmar Lapp, Wasila Dahdul, Josef Uyeda.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.5.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
