<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>RPhenoscape Tutorial • SCATE.shortcourse</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="RPhenoscape Tutorial">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SCATE.shortcourse</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/01_scate-shortcourse.html">RPhenoscape Tutorial</a>
    </li>
    <li>
      <a href="../articles/PARAMO.nb.html">PARAMO Tutorial</a>
    </li>
    <li>
      <a href="../articles/glossary.html">Glossary</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/phenoscape/scate-shortcourse">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>RPhenoscape Tutorial</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/phenoscape/scate-shortcourse/blob/master/vignettes/01_scate-shortcourse.Rmd"><code>vignettes/01_scate-shortcourse.Rmd</code></a></small>
      <div class="hidden name"><code>01_scate-shortcourse.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>1. Introduction</h2>
<p>The desired input of many comparative studies is a matrix that includes all the data available for a set of traits and a focal taxon. Bringing similar types of trait data together across studies is notoriously difficult, but is straightforward with ontology annotations. Trait data that are tagged using similar ontology terms can be automatically called using R functions.</p>
<p>This workshop uses trait data for vertebrates that has been annotated (tagged) with ontology terms. These data are accessed from the Phenoscape Knowledgebase (KB). In this lesson we use R functions to obtain and view character matrices from the KB, obtain synthetic matrices of inferred presence/absence characters, and understand the meaning and relations of ontology terms. The lesson assumes basic knowledge of R and RStudio.</p>
</div>
<div id="browse-and-filter-data-in-the-kb-to-understand-the-scope-of-available-data-" class="section level2">
<h2 class="hasAnchor">
<a href="#browse-and-filter-data-in-the-kb-to-understand-the-scope-of-available-data-" class="anchor"></a>2. Browse and filter data in the KB to understand the scope of available data.</h2>
<p>You would like to understand the evolution of morphological traits in a group, and you need a matrix to map to your (typically molecular) phylogenetic tree. You come to the Phenoscape KB to see what types of data are available across the members of your clade. In this exercise, you are an ichthyologist, interested in catfishes (Siluriformes), specifically the bullhead catfishes (Ictaluridae).</p>
<p>You begin by querying ‘Siluriformes’ in the Phenoscape KB faceted browsing page (<a href="http://beta.phenoscape.org/#/facet" class="uri">http://beta.phenoscape.org/#/facet</a>). You immediately see the scope of the data:</p>
<p>From left to right across the top tabs, for ‘Siluriformes’:</p>
<ul>
<li>There are 4,537 unique phenotypes</li>
<li>There are 292,356 taxon annotations</li>
<li>There are 1,452 siluriform taxa with data in the KB</li>
<li>There are 25 publications in the KB that include catfish phenotypes</li>
<li>There are 17,883 candidate genes linked to the phenotypes</li>
</ul>
<p>Taxon annotations are the ontology terms tagged to free text descriptions of phenotypes for each siluriform taxon. Click on the ‘Sources’ box on the right hand side of each row to see the publication source(s) and free text description of the phenotype to which each ontology annotation was tagged.</p>
<p>You are specifically interested in Ictaluridae (bullhead catfishes), and you click on that family in the faceted browsing interface. Now filter using ‘fin’ as the anatomical entity and click ‘parts’.</p>
<p>Download these taxa and their morphological trait data using the link provided.</p>
<p>Go to the publications tab, which shows nine publications that contain fin data. Note that any of these studies, e.g. Lundberg (1992), can be entered under ‘publications’ in faceted browsing, but the original matrix cannot be downloaded.</p>
</div>
<div id="retrieve-term-info-using-rphenoscape" class="section level2">
<h2 class="hasAnchor">
<a href="#retrieve-term-info-using-rphenoscape" class="anchor"></a>3. Retrieve term info using RPhenoscape</h2>
<p>Now we will repeat some of the above steps in R, beginning with a query for ‘Ictaluridae’ and ‘fin’ (including parts) to retrieve the list of studies that contain fin characters for this taxon.</p>
<p>The <a href="http://rphenoscape.phenoscape.org/">RPhenoscape</a> package provides convenient access to the Phenoscape Knowledgebase in the form of R functions returning R-native data structures. So we don’t have to prefix every call of the package’s functions with the package name, we will first attach the package:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(rphenoscape)</code></pre></div>
<p>First let’s look up information about the query terms. You can use the term detail functions <code><a href="http://rphenoscape.phenoscape.org/reference/pk_terms.html">pk_anatomical_detail()</a></code>, <code><a href="http://rphenoscape.phenoscape.org/reference/pk_terms.html">pk_taxon_detail()</a></code>, <code><a href="http://rphenoscape.phenoscape.org/reference/pk_terms.html">pk_phenotype_detail()</a></code>, and <code><a href="http://rphenoscape.phenoscape.org/reference/pk_terms.html">pk_gene_detail()</a></code> to retrieve an ontology term’s ID, label, and definition.</p>
<p>For example, view the details for the anatomy term “fin”:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://rphenoscape.phenoscape.org/reference/pk_terms.html">pk_anatomical_detail</a></span>(<span class="st">"fin"</span>)
<span class="co">#&gt;   label</span>
<span class="co">#&gt; 1   fin</span>
<span class="co">#&gt;                                                                                           definition</span>
<span class="co">#&gt; 1 An external projection of an aquatic animal as a fish used in propelling or guiding the body[BTO].</span>
<span class="co">#&gt;                                              id</span>
<span class="co">#&gt; 1 http://purl.obolibrary.org/obo/UBERON_0008897</span></code></pre></div>
<p>For a specific taxon, such as Siluriformes, you can view the taxon rank, term label, whether or not the taxon is extinct, its common name, and ID:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://rphenoscape.phenoscape.org/reference/pk_terms.html">pk_taxon_detail</a></span>(<span class="st">"Siluriformes"</span>)
<span class="co">#&gt;                                          rank.id rank.label        label</span>
<span class="co">#&gt; 1 http://purl.obolibrary.org/obo/TAXRANK_0000003      order Siluriformes</span>
<span class="co">#&gt;   extinct common_name                                         id</span>
<span class="co">#&gt; 1   FALSE     catfish http://purl.obolibrary.org/obo/VTO_0034991</span></code></pre></div>
<p>Ontology terms are related to each other by superclass (parent) and subclass (child) relationships, among other relationships such as part_of. You can use the term classification functions <code><a href="http://rphenoscape.phenoscape.org/reference/pk_class.html">pk_anatomical_class()</a></code>, <code><a href="http://rphenoscape.phenoscape.org/reference/pk_class.html">pk_taxon_class()</a></code>,and <code><a href="http://rphenoscape.phenoscape.org/reference/pk_class.html">pk_phenotype_class()</a></code>to view the superclass and subclass relationships for anatomy, taxonomy, or phenotypic quality terms.</p>
<p>For example, view the relationships for “fin”:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://rphenoscape.phenoscape.org/reference/pk_class.html">pk_anatomical_class</a></span>(<span class="st">"fin"</span>)
<span class="co">#&gt; Querying the IRI for fin</span>
<span class="co">#&gt; Retrieving classification information</span>
<span class="co">#&gt; $label</span>
<span class="co">#&gt; [1] "fin"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $subClassOf</span>
<span class="co">#&gt;                                             @id     label</span>
<span class="co">#&gt; 1 http://purl.obolibrary.org/obo/UBERON_0000026 appendage</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $equivalentTo</span>
<span class="co">#&gt; list()</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $superClassOf</span>
<span class="co">#&gt;                                                                                                                                                                                                                                                                                          @id</span>
<span class="co">#&gt; 1                                                                                                                                                                                                                                                 http://purl.obolibrary.org/obo/XAO_0000002</span>
<span class="co">#&gt; 2                                                                                                                                                                                                                                                 http://purl.obolibrary.org/obo/ZFA_0000108</span>
<span class="co">#&gt; 3  http://purl.org/phenoscape/subclassof?value=%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0008897%3E%0A+and+%28%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FBFO_0000050%3E+some+%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0003101%3E%29#0d8e50eb-746d-4198-8524-aff0ac73d678</span>
<span class="co">#&gt; 4  http://purl.org/phenoscape/subclassof?value=%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0008897%3E%0A+and+%28%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FBFO_0000050%3E+some+%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0003101%3E%29#9564f553-a93a-4495-899e-12f233824168</span>
<span class="co">#&gt; 5  http://purl.org/phenoscape/subclassof?value=%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0008897%3E%0A+and+%28%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FBFO_0000050%3E+some+%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0003101%3E%29#b33fe59f-e303-49f2-bb51-0f150cffd1dd</span>
<span class="co">#&gt; 6  http://purl.org/phenoscape/subclassof?value=%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0008897%3E%0A+and+%28%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FBFO_0000050%3E+some+%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0003101%3E%29#beb558ba-1e78-4ae2-95f5-aba3485f2f1e</span>
<span class="co">#&gt; 7  http://purl.org/phenoscape/subclassof?value=%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0008897%3E%0A+and+%28%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FBFO_0000050%3E+some+%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0003101%3E%29#0336e97b-4dbf-4931-893a-70396326b38b</span>
<span class="co">#&gt; 8  http://purl.org/phenoscape/subclassof?value=%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0008897%3E%0A+and+%28%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FBFO_0000050%3E+some+%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0003101%3E%29#eeef048f-3018-4a7d-b879-771b6f71164e</span>
<span class="co">#&gt; 9  http://purl.org/phenoscape/subclassof?value=%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0008897%3E%0A+and+%28%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FBFO_0000050%3E+some+%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0003101%3E%29#70836752-db49-4ef8-a6f9-d4cf858ec4c7</span>
<span class="co">#&gt; 10 http://purl.org/phenoscape/subclassof?value=%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0008897%3E%0A+and+%28%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FBFO_0000050%3E+some+%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0003101%3E%29#46acf662-5103-4753-8219-ed52e0362e47</span>
<span class="co">#&gt; 11                                                                                                                                                                                                                                             http://purl.obolibrary.org/obo/UBERON_4000162</span>
<span class="co">#&gt; 12                                                                                                                                                                                                                                             http://purl.obolibrary.org/obo/UBERON_0002534</span>
<span class="co">#&gt;                                   label</span>
<span class="co">#&gt; 1                            fin (frog)</span>
<span class="co">#&gt; 2                       fin (zebrafish)</span>
<span class="co">#&gt; 3  fin and (part of some male organism)</span>
<span class="co">#&gt; 4  fin and (part of some male organism)</span>
<span class="co">#&gt; 5  fin and (part of some male organism)</span>
<span class="co">#&gt; 6  fin and (part of some male organism)</span>
<span class="co">#&gt; 7  fin and (part of some male organism)</span>
<span class="co">#&gt; 8  fin and (part of some male organism)</span>
<span class="co">#&gt; 9  fin and (part of some male organism)</span>
<span class="co">#&gt; 10 fin and (part of some male organism)</span>
<span class="co">#&gt; 11                           median fin</span>
<span class="co">#&gt; 12                           paired fin</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`@id`</span>
<span class="co">#&gt; [1] "http://purl.obolibrary.org/obo/UBERON_0008897"</span></code></pre></div>
</div>
<div id="retrieve-study-matrices-in-rphenoscape" class="section level2">
<h2 class="hasAnchor">
<a href="#retrieve-study-matrices-in-rphenoscape" class="anchor"></a>4. Retrieve study matrices in RPhenoscape</h2>
<p>Now we will query for ‘Ictaluridae’ and ‘fin’ (including parts) to retrieve the list of studies that contain fin characters for this taxon.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">slist &lt;-<span class="st"> </span><span class="kw"><a href="http://rphenoscape.phenoscape.org/reference/pk_get_study_list.html">pk_get_study_list</a></span>(<span class="dt">taxon =</span> <span class="st">"Ictaluridae"</span>, <span class="dt">entity =</span> <span class="st">"fin"</span>, <span class="dt">relation =</span> <span class="st">"part of"</span>)</code></pre></div>
<p>The list is returned as a data.frame containing the study IDs and citations. Navigate to Global Environment and click on ‘slist’ to view the ID and citation (author, year) for each study in the KB that contains any ictalurid species associated with fin data. Note that taxon, entity, and quality are optional arguments in this query, and we have also included fin parts. Here we found nine studies.</p>
<p>Next we want to obtain the character matrix for one of these studies. Using the study IDs from the previous step, we can get a list of the character matrices as NeXML objects. We will choose the Lundberg (1992) publication, the fifth item on the list above.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">lundberg_nexml &lt;-<span class="st"> </span><span class="kw"><a href="http://rphenoscape.phenoscape.org/reference/pk_get_study_xml.html">pk_get_study_xml</a></span>(slist<span class="op">$</span>id[<span class="dv">5</span>])
<span class="co">#&gt; ....This might take a while....</span>
<span class="co">#&gt; https://scholar.google.com/scholar?q=The+Phylogeny+of+Ictalurid+Catfishes%3A+A+Synthesis+of+Recent+Work&amp;btnG=&amp;hl=en&amp;as_sdt=0%2C42</span></code></pre></div>
<p>Now go to Global Environment pane and click on ‘lundberg_nexml’ to view the study ID and associated NeXML object.</p>
<p>To view the full Lundberg (1992) matrix within RStudio, we need to first retrieve the matrix as a data.frame from the NeXML object:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">lundberg_matrix &lt;-<span class="st"> </span><span class="kw"><a href="http://rphenoscape.phenoscape.org/reference/pk_get_study.html">pk_get_study</a></span>(lundberg_nexml)
<span class="co">#&gt; Map symbols to labels...</span></code></pre></div>
<p>You can take a look at a small part of the Lundberg (1992) matrix (e.g., the first five taxa and first five characters):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">lundberg_matrix[[<span class="dv">1</span>]][<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">7</span>]
<span class="co">#&gt;                 taxa                                      otu</span>
<span class="co">#&gt; 1  Ameiurus brunneus otu_84e7fa13-6c3f-4f51-8448-9fe7c4310c19</span>
<span class="co">#&gt; 2     Ameiurus catus otu_1b65aadf-15b3-4949-a6f2-3d0f3ad148d9</span>
<span class="co">#&gt; 3     Ameiurus melas otu_a1ae820b-afe8-4356-9eac-56d3d3da2cde</span>
<span class="co">#&gt; 4   Ameiurus natalis otu_65987953-4897-4a76-b789-506a3c56010c</span>
<span class="co">#&gt; 5 Ameiurus nebulosus otu_e95287c2-2f9d-4d6f-8145-f0c35dfe524e</span>
<span class="co">#&gt;   Anal-fin rays, species mean count Anterior dentations of pectoral spine</span>
<span class="co">#&gt; 1                           16.1-20                                 large</span>
<span class="co">#&gt; 2                         20.1-25.5                              moderate</span>
<span class="co">#&gt; 3                         20.1-25.5                                 small</span>
<span class="co">#&gt; 4                           25.6-28                                 small</span>
<span class="co">#&gt; 5                         20.1-25.5                              moderate</span>
<span class="co">#&gt;   Anterior distal serrae of pectoral spine</span>
<span class="co">#&gt; 1               &lt;3 moderately sharp serrae</span>
<span class="co">#&gt; 2              3-6 moderately sharp serrae</span>
<span class="co">#&gt; 3             absent or scarcely developed</span>
<span class="co">#&gt; 4              3-6 moderately sharp serrae</span>
<span class="co">#&gt; 5               &lt;3 moderately sharp serrae</span>
<span class="co">#&gt;                                                    Anterior extent of sphenotic</span>
<span class="co">#&gt; 1 reaches or extends anterior to the level of anterior margin of epiphyseal bar</span>
<span class="co">#&gt; 2 reaches or extends anterior to the level of anterior margin of epiphyseal bar</span>
<span class="co">#&gt; 3 reaches or extends anterior to the level of anterior margin of epiphyseal bar</span>
<span class="co">#&gt; 4 reaches or extends anterior to the level of anterior margin of epiphyseal bar</span>
<span class="co">#&gt; 5 reaches or extends anterior to the level of anterior margin of epiphyseal bar</span>
<span class="co">#&gt;                     Anterior limb of fourth transverse process</span>
<span class="co">#&gt; 1 supracleithral facet shallow, basal recess shallow and broad</span>
<span class="co">#&gt; 2       supracleithral facet deep, basal recess deep and broad</span>
<span class="co">#&gt; 3       supracleithral facet deep, basal recess deep and broad</span>
<span class="co">#&gt; 4      supracleithral facet deep, basal recess deep and narrow</span>
<span class="co">#&gt; 5      supracleithral facet deep, basal recess deep and narrow</span></code></pre></div>
<p>Now navigate to Global Environment and click on ‘lundberg_matrix’ to view the matrix, including the total number of taxa and characters (i.e., the dimensions of the data.frame). To view the entire matrix in this panel, click on the first icon that appears to the right when you hover over the second row.</p>
<p>Note that all taxa and characters from this study are returned, not just those pertaining to our original search terms. In this example, many anatomical entities other than ‘fin’ and its parts are returned as part of the original study matrix. In a later step, we will learn how to subset a matrix using anatomy or taxonomy terms.</p>
</div>
<div id="get-a-presenceabsence-character-matrix-using-ontotrace-in-rphenoscape" class="section level2">
<h2 class="hasAnchor">
<a href="#get-a-presenceabsence-character-matrix-using-ontotrace-in-rphenoscape" class="anchor"></a>5. Get a presence/absence character matrix using OntoTrace in RPhenoscape</h2>
<p>In the previous step, we queried the KB for individual studies that contain characters pertaining to the fins in ictalurid catfish, and we obtained character matrices for source publications that included those characters. Now we want to get a matrix that synthesizes these characters into a single matrix.</p>
<p><a href="http://beta.phenoscape.org/#/ontotrace">OntoTrace</a> from the KB combines characters from multiple publications into a single synthetic matrix. It allows a user to specify any number or combination of anatomical entities and taxa. It returns a matrix as a nexml object that includes only presence or absence of anatomical entities. It uses asserted data of presence or absence, i.e., originally described by an author, or inferred data. Data are inferred from asserted characters, e.g., if an author describes a part of a fin such as ‘fin ray’, the fin is inferred present and scored as such in the matrix.</p>
<p>The OntoTrace default settings are to include parts of the anatomical entity and to include variable-only characters, but here we will include those that are invariant as well.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nex &lt;-<span class="st"> </span><span class="kw"><a href="http://rphenoscape.phenoscape.org/reference/pk_get_ontotrace_xml.html">pk_get_ontotrace_xml</a></span>(<span class="dt">taxon =</span> <span class="st">"Ictaluridae"</span>, <span class="dt">entity =</span> <span class="st">"fin"</span>, <span class="dt">variable_only =</span> <span class="ot">FALSE</span>)</code></pre></div>
<p>To view the matrix within RStudio, get the matrix (m) as a <code>data.frame</code> from the NeXML object. Opening this (‘m’) in the Global Environment shows the taxa, character names, and state assignments (0, absent; 1, present) (note: the otu and otus are NeXML-internal IDs):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">m &lt;-<span class="st"> </span><span class="kw"><a href="http://rphenoscape.phenoscape.org/reference/pk_ontotrace.html">pk_get_ontotrace</a></span>(nex)</code></pre></div>
<p>The character and taxon lists for the matrix can also be returned. Click on ‘meta’ in the Global Environment pane to view the data.frames for taxa and entities:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">meta &lt;-<span class="st"> </span><span class="kw"><a href="http://rphenoscape.phenoscape.org/reference/pk_ontotrace.html">pk_get_ontotrace_meta</a></span>(nex)</code></pre></div>
<p>The above returns 29 species of ictalurid catfishes and 123 anatomical entities for fins and their parts.</p>
<p>You will need to go the <a href="http://beta.phenoscape.org/#/facet">Phenoscape KB</a> to view the supporting states, i.e., the source data upon which inferences were made or those that were directly asserted.</p>
<p>For example, in the returned OntoTrace matrix (see ‘m’ in Global Environment (row 10)) the pelvic fin is scored as present for <em>Ictalurus australis</em>. Enter these data into the faceted browsing interface:</p>
<ul>
<li>Anatomical entity = ‘pelvic fin’</li>
<li>Quality = ‘present’</li>
<li>Taxon = ‘Ictalurus australis’</li>
</ul>
<p>Looking at ‘Taxon annotations’, you will find that there are zero phenotypes, i.e., no asserted character states to ‘pelvic fin present’. Now click the ‘inferred presence’ radio button under Phenotypic quality. You will now see three phenotypes for three taxa in the KB.</p>
</div>
<div id="subset-the-matrix-taxonomically" class="section level2">
<h2 class="hasAnchor">
<a href="#subset-the-matrix-taxonomically" class="anchor"></a>6. Subset the matrix taxonomically</h2>
<p>In the above step you retrieved a synthetic matrix for the Ictaluridae. Now you would like to pare it down in taxonomic and anatomical scope. Let’s begin by trimming the matrix down to the presence and absence characters pertaining only to the genus Ictalurus. We will reduce the matrix to include only descendents of Ictalurus.</p>
<p>First, we need to parse the list to determine which taxa in the matrix (‘m’) are descendants of <em>Ictalurus</em>. This returns a list of TRUE or FALSE assignments for each taxon in the matrix:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">is_desc &lt;-<span class="st"> </span><span class="kw"><a href="http://rphenoscape.phenoscape.org/reference/pk_is_descendant.html">pk_is_descendant</a></span>(<span class="st">'Ictalurus'</span>, m<span class="op">$</span>taxa)</code></pre></div>
<p>We then subset the matrix m to the descendants of <em>Ictalurus</em>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ictalurus_m &lt;-<span class="st"> </span>m[is_desc, ]</code></pre></div>
<p>View the new matrix by clicking on ‘ictalurus_m’ in the Global Environment to view the <code>data.frame</code> of the matrix.</p>
<p>We can also subset the matrix by anatomy. In this case, let’s select characters for the pectoral fin and its parts. We will use <code><a href="http://rphenoscape.phenoscape.org/reference/pk_is_descendant.html">pk_is_descendant()</a></code> to first parse the characters (column names) to determine which are descendents of the term ‘pectoral fin’. Because this function only looks for descendent terms, we also need to specify that any characters with ‘pectoral fin’ are included, and we exclude the first three taxon-related columns:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">is_desc_pecfin &lt;-<span class="st"> </span><span class="kw"><a href="http://rphenoscape.phenoscape.org/reference/pk_is_descendant.html">pk_is_descendant</a></span>(<span class="st">'pectoral fin'</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(m)[<span class="op">-</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>)], <span class="dt">includeRels =</span> <span class="st">"part_of"</span>) <span class="op">|</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(m)[<span class="op">-</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>)] <span class="op">==</span><span class="st"> 'pectoral fin'</span> 
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(is_desc_pecfin)
<span class="co">#&gt; is_desc_pecfin</span>
<span class="co">#&gt; FALSE  TRUE </span>
<span class="co">#&gt;   109    14</span></code></pre></div>
<p>View the character names to verify that pectoral fin related characters will be selected:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(m)[<span class="op">-</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>)][is_desc_pecfin]
<span class="co">#&gt;  [1] "anterior dentation of pectoral fin spine"       </span>
<span class="co">#&gt;  [2] "anterior distal serration of pectoral fin spine"</span>
<span class="co">#&gt;  [3] "pectoral fin"                                   </span>
<span class="co">#&gt;  [4] "pectoral fin lepidotrichium"                    </span>
<span class="co">#&gt;  [5] "pectoral fin proximal radial bone"              </span>
<span class="co">#&gt;  [6] "pectoral fin proximal radial cartilage"         </span>
<span class="co">#&gt;  [7] "pectoral fin proximal radial element"           </span>
<span class="co">#&gt;  [8] "pectoral fin radial bone"                       </span>
<span class="co">#&gt;  [9] "pectoral fin radial cartilage"                  </span>
<span class="co">#&gt; [10] "pectoral fin radial element"                    </span>
<span class="co">#&gt; [11] "pectoral fin radial skeleton"                   </span>
<span class="co">#&gt; [12] "pectoral fin skeleton"                          </span>
<span class="co">#&gt; [13] "pectoral fin spine"                             </span>
<span class="co">#&gt; [14] "posterior dentation of pectoral fin spine"</span></code></pre></div>
<p>We now subset the matrix m using ‘is_desc_pecfin’ and retaining the first taxon-related columns:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">to_keep &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="ot">TRUE</span>, <span class="ot">TRUE</span>, <span class="ot">TRUE</span>, is_desc) 
pecfin_m &lt;-<span class="st"> </span>m[, to_keep]</code></pre></div>
<p>View this matrix of pectoral fin characters for Ictaluridae by clicking on ‘pecfin_m’ in the Global Environment.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">1. Introduction</a></li>
      <li><a href="#browse-and-filter-data-in-the-kb-to-understand-the-scope-of-available-data-">2. Browse and filter data in the KB to understand the scope of available data.</a></li>
      <li><a href="#retrieve-term-info-using-rphenoscape">3. Retrieve term info using RPhenoscape</a></li>
      <li><a href="#retrieve-study-matrices-in-rphenoscape">4. Retrieve study matrices in RPhenoscape</a></li>
      <li><a href="#get-a-presenceabsence-character-matrix-using-ontotrace-in-rphenoscape">5. Get a presence/absence character matrix using OntoTrace in RPhenoscape</a></li>
      <li><a href="#subset-the-matrix-taxonomically">6. Subset the matrix taxonomically</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Hilmar Lapp, Wasila Dahdul, Josef Uyeda.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
